reana_job_controller/job_monitor.py

```diff
@@ -133,7 +150,9 @@ def watch_jobs(self, job_db, app=None):
                     logging.info('Grabbing pod {} logs...'.format(
                         last_spawned_pod.metadata.name))
                     job_db[job_id]['log'] = \
-                        self.get_container_logs(last_spawned_pod)
```

Comment:
I think we don't need these lines anymore (see below, `LN 143-151`) since the "last pod" is already the one you have in memory inside the `job` variable:
```python
                    logging.info('Getting last spawned pod for kubernetes'
                                 ' job {}'.format(kubernetes_job_id))
                    last_spawned_pod = \
                        current_k8s_corev1_api_client.list_namespaced_pod(
                            namespace=job.metadata.namespace,
                            label_selector='job-name={job_name}'.format(
                                job_name=kubernetes_job_id)).items[-1]
                    logging.info('Grabbing pod {} logs...'.format(
                        last_spawned_pod.metadata.name))
```
What do you think @roksys? If you think the same I will create an issue and we address this later (we would save Kubernetes API server from making yet another REST call).